processing/src/main/java/org/apache/druid/query/aggregation/last/StringLastVectorAggregator.java

```diff
@@ -123,13 +123,14 @@ public void aggregate(
   {
     long[] timeVector = timeSelector.getLongVector();
     Object[] objectsWhichMightBeStrings = valueSelector.getObjectVector();
+    // one time check on first element to judge if elements are serializable pairs or not
+    final int startRow = rows == null ? 0 : rows[0];
+    final boolean foldNeeded = StringFirstLastUtils.objectNeedsFoldCheck(objectsWhichMightBeStrings[startRow]);
```

Comment:
Similar comment to line 77

Reply:
After the change, we iterate over the array of objects to find the first non-null object and call it on that object only. That value is used at the rest of the places. This ensures similar behavior to the other case as above